File: cpp/src/board.cpp
#include <cstdlib>
#include <ctime>
#include <iostream>
#include <vector>

#include "board.hpp"

Board::Board(int height, int width) {
    this->height = height;
    this->width = width;
    this->cells = std::vector<bool>();

    std::srand(std::time(nullptr));
    for (int i = 0; i < height * width; i++) {
        this->cells.push_back(std::rand() % 2 > 0.5);
    }
}

Board::Board(const Board &other) {
    this->height = other.height;
    this->width = other.width;
    this->cells = std::vector<bool>(other.cells);
}

bool Board::getCell(const int &row, const int &col) {
    if (row >= 0 && col >= 0 && row < this->height && col < this->width) {
        return this->cells[(row * this->width) + col];
    } else {
        return false;
    }
}

void Board::draw() {
    for (int i = 0; i < this->height; i++) {
        for (int j = 0; j < this->width; j++) {
            if (this->cells[(i * this->width) + j]) {
                std::cout << "██";
            } else {
                std::cout << "  ";
            }
        }
        std::cout << std::endl;
    }
}

File: python/board.py
import random
from typing import List
from io import StringIO

# import copy


class Board:
    def __init__(self, height: int, width: int, cells: List[bool] = [], swap: List[bool] = []):
        self.cells: List[bool] = [bool(random.getrandbits(1)) for _ in range(height * width)] if len(cells) == 0 else cells
        self.swap: List[bool] = [False] * (height * width) if len(swap) == 0 else swap
        self.height = height
        self.width = width

    def draw(self, buf: StringIO) -> StringIO:
        for i in range(self.height):
            for j in range(self.width):
                if self.cells[(i * self.width) + j]:
                    buf.write("██")
                else:
                    buf.write("  ")
            buf.write("\n")
        return buf

    def get_cell(self, row: int, col: int) -> int:
        if 0 <= row < self.height and 0 <= col < self.width:
            return self.cells[(row * self.width) + col]
        else:
            return False

    @property
    def size(self) -> int:
        return self.height * self.width
